### ISSUE TITLE
[Class] Mage

### ISSUE THREAD

**Issue by vexation47** [WoWManiaUK/Redemption#654]
So I am here to post a recap of what is currently wrong with mages, and some have been present for a while.  Its almost as though Mages are getting no love for some biased reason.

1- Hot Streak! is consumed upon impact.  This may not seem like a big deal, but when optimizing dps it is.  It needs to be consumed upon use.  Otherwise I cannot save a hot streak until the T10 2 set bonus is over, lowering the up time on Pushing the Limit.  The objective a TTW/Fire Mage should strive for is to be able to fit as many fireballs as they can inside pushing the limit as well as keeping Pushing the Limit up as much as possible.   With Hot Streak being consumed upon impact you lose your next hot streak if you crit again before the pyro hits the target

2- Speaking of Pushing the Limit, it no longer activates after Hot Streak is used, so it makes the BiS gear, not so BiS anymore, its not just a small dps loss either if you play a mage right, because as stated above the goal is to try to keep up the 12% haste from pushing the limit as much as possible.  I noticed it is still working fine for brainfreeze.

3- Combustion - It is supposed to have 3 non periodic crits to expire, however for some reason it seems fireball and pyroblast crits are consider 2 crits for this talent.  The talent being tested with scroch works fine, but when you add pyro or fireball then you only are able to get 2 crits in a combustion.

4- Mirror Images - It sort of works....like if you are out questing, and use it, the targets hit the images before you, however it is also supposed to hide your threat, It is supposed to be a great tool for managing your threat, start the fight with it and the tank can build agro while you unload on the boss with images up.  You are supposed to get the threat back once its over however.

5- Fingers of Frost - Both charges are consumed upon 1 spell hit making it hard to get a shatter combo from it. If it was blizzlike then I should be able to do 2 frostbolts and chain an icelance off the last one and all 3 get the shatter bonus. (since the icelance doesn't recognize that the 2nd frost bolt hit yet.

6- Here is another thing that I am telling you out of the kindness of my heart, as I love grinding with this current bug.  When you use brainfreeze it not only activates the mage t10 2 set bonus, but also activates the 4 set bonus without having to use mirror images.  So 18% damage up almost all the time? Nice!

7 - I also think Arcane Missiles are still broken as I see no arcane mages anymore.  It was only getting 2 hits off the cast and not benefiting from missile barrage rendering the arcane spec useless besides dipping in for TTW.

So in summary, its not fun to be a mage right now.

**Comment by vexation47** [comment]
Oh and....

8- Frost Nova (and other frozen effects) - They break from any damage, its not supposed to break 100% of the time, it MAY break.  I am not sure the blizzlike mechanics but I do recall from experience a frost bolt crit usually broke it, but I do ALWAYS remember fighting warriors, and as a safety measure (as a frost mage) I would always keep up fire ward so I can fire blast the spell reflect.  So when I freeze them I usually open with a fireblast since a warrior only really has the option to spellreflect at that time.  Sometimes they don't but then I can chain a deepfreeze after, on this server I cant because 100% of the time, fireblast breaks the freeze.

**Comment by GMWeeble** [comment]
Mirror image does hide your threat. Works just fine, but the Images are stacked not spread out around you.

**Comment by vexation47** [comment]
I can confirm all of these with fraps if you'd like.  I pulled off the tanks several times in ICC with images up.  Perhaps it has a threshold, or only hides threat from certain spells, because it only really happens when I'm doing 20k+ dps on trash spamming living bomb, flamestrike and blizzard.

**Comment by GMWeeble** [comment]
If you  are using it on tras if might bug out. But It works just fine on Single target.

**Comment by philoudu16** [comment]
What I think may happen with multiple mobs, is it hides the threat for mobs you have already damaged or entered the threat table in some other way before you cast Mirror Image (at least, that seems to be how it works when you're soloing things). I don't know how it's supposed to work on retail though :)

**Comment by vexation47** [comment]
Yeah, I know whats wrong with it now, anything that enters combat after you use images ignores it, its not supposed to happen like that, but I can live with it.

**Comment by Emcitementz** [comment]
> Yeah, I know whats wrong with it now, anything that enters combat after you use images ignores it, its not supposed to happen like that, but I can live with it.

Well, mirror images has already been logged and something has been changed for it, your actual threat is lost when the images disappear.
So, what's your saying is the opposite of what has been said there -> [#574](https://github.com/WoWManiaUK/Redemption/issues/574), so please before requesting changes, make sure that it has to be like this or like that.

**Comment by vexation47** [comment]
A- 574 is closed
B- I never actually thought this post would of just been closed immediately as I was kind of snarky and not specifying one thing.
C- You are supposed to get your threat back after images is over...its how it worked in retail.  It didn't eliminate threat.  It works kind of like the opposite of mocking blow, mocking blow doesn't taunt the boss, it just makes them attack you for so long allowing you to build threat during that time, so if your threat is good after mocking blow, the boss stays on you.   As for images, it is supposed to be used at the begining of a fight for example.  It just hides the threat you are doing, allowing you to just give it your all, while the tank builds enough threat.   If the tank doesn't build enough threat, then you pull the boss when images expire.  That is how blizzard intended it to work.

**Comment by Emcitementz** [comment]
Hot Streak will now be consumed when pyroblast is casted -> Fixed in dev server.
Fingers of frost mechanic has also been fixed in dev server.
The rest of bugs is still in development.

**Comment by Emcitementz** [comment]
Missile barrage has been fixed.
Mage T10 4P as well.

**Comment by Emcitementz** [comment]
Mage T10 2P fixed in dev.

**Comment by vexation47** [comment]
Thanks :) I can confirm they are all indeed fixed, besides Missile Barrage, I'm rarely Arcane.

**Comment by vexation47** [comment]
Actually T10 2 set was working fine last night, but its not working again to day :(

**Comment by Emcitementz** [comment]
It's all working on dev.

**Comment by Silva69** [comment]
T10 2 for arcanes mages is working but is not working a 100%, the way of working is different from arcane to fire mage,  because the bonus "pushing the limits" (12% for 5secs) is proc early. we only have to get that proc when you finish cast arcane missiles with the proc missile barrage, not at the begining, check this video from retail and you see what im talking, just see when the bonus proc and compare

https://www.youtube.com/watch?v=jD4Ds6f_6wo
